thread/common/omrthread.c

```diff
@@ -679,11 +679,17 @@ init_spinParameters(omrthread_library_t lib)
  *
  * @see omrthread_init
  */
-void
+static void
 omrthread_shutdown(void)
 {
 	omrthread_library_t lib = GLOBAL_DATA(default_library);
 	ASSERT(lib);
+	/* Acquire the global lock here to wait until all outstanding omrthread_exit calls
+	 * have completed. omrthread_exit holds the global lock until the thread is ready
+	 * to exit to the OS.
+	 */
```

Comment:
The goal is to wait here until all outstanding `omrthread_exit` calls have completed. `GLOBAL_LOCK_SIMPLE` uses a mutex. A mutex won't guarantee that all `omrthread_exit` calls have completed. Shouldn't a condition variable be used over here?

Reply:
I only care about omrthread_exit calls that were in progress when the shutdown begins. Any thread library call issued after shutdown begins is going to crash/fail randomly.

Reply:
The correct pattern is to inform all your forked threads of the need to exit, and wait for them to indicate that with a monitor protecting a count variable or somesuch. Each thread calls omrthread_exit with the monitor after indicating completion. Once they've all checked in, the library shutdown can begin. This code ensures that each thread executing omrthread_exit has reached the point of OS termination (and never needs to reference the thread library structures again). In pratice, only one thread could be in this state when shutdown begins.